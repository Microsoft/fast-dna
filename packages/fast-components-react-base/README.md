# FAST Components React Base
A set of React components without styles.

These can be combined with JSS styles to create styled components or used as-is.

## Installation
`npm i --save @microsoft/fast-components-react-base`

## Usage
An example of using one of the components from the `@microsoft/fast-components-react-base` package:

```
import React from "react";
import ReactDOM from "react-dom";
import { Button } from "@microsoft/fast-components-react-base";

const root = document.createElement("div");
root.setAttribute("id", "root");
document.body.appendChild(root);

function render(): void {
    ReactDOM.render(
        <Button primary={true}>
            Click me!
        </Button>,
        root
    );
}

render();
```

An example of using the `manageJss` from `@microsoft/fast-jss-manager-react` higher order component to export a component with a JSS style. See the `@microsoft/fast-components-class-name-contracts-base` for the list of class names that can be applied to the base components.

```
import React from "react";
import { Button } from "@microsoft/fast-components-react-base";
import manageJss from "@microsoft/fast-jss-manager-react";

const buttonStyles = {
    button: {
        background: "blue"
    }
};

export default manageJss(buttonStyles)(Button);
```

## Component documentation
Each component comes with a README.md for implementation and other usage details, a JSON schema for the properties that can be used and additional typescript definition files.
