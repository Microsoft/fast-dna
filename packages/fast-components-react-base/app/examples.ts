import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };
<<<<<<< HEAD

import DividerExamples from "../src/divider/examples.data";
export { DividerExamples };

import ImageExamples from "../src/image/examples.data";
export {ImageExamples};

import ToggleExamples from "../src/toggle/examples.data";
export { ToggleExamples };

import TypographyExamples from "../src/typography/examples.data";
export { TypographyExamples };

import HyperlinkExamples from "../src/hyperlink/examples.data";
export { HyperlinkExamples };
=======
export {ButtonExamples};

import HypertextExamples from "../src/hypertext/examples.data";
export {HypertextExamples};
>>>>>>> renamed all to hypertext from hyperlink
