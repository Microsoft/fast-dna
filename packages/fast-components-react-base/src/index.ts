import Button from "./button";
export { Button };
export * from "./button";

import Divider from "./divider";
export { Divider };
export * from "./divider";

import Toggle from "./toggle";
export {Toggle};
export * from "./toggle";

import Foundation from "./foundation";
export { Foundation };
export * from "./foundation";
