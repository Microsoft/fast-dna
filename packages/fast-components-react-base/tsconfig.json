{
    "compilerOptions": {
        "moduleResolution": "node",
        "resolveJsonModule": true,
        "esModuleInterop": true,
        "module": "commonjs",
        "target": "ES6",
        "baseUrl": ".",
        "declaration": true,
        "jsx": "react",
        "outDir": "./dist",
        "noEmit": true
    },
    "include": [
        "src/**/*"
    ],
    "exclude": [
        "node_modules",
        "**/*.spec.ts",
        "**/*.spec.tsx"        
    ]
}
