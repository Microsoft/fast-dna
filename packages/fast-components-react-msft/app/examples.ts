import ButtonExamples from "../src/button/examples.data";
export {ButtonExamples};

import MediaExamples from "../src/media/examples.data";

export {MediaExamples};

import DividerExamples from "../src/divider/examples.data";
export {DividerExamples};
