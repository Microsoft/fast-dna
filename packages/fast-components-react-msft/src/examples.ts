import * as React from "react";

export type IExampleData<T> = T;

export interface IGenericExample<T> {
    name: string;

    component: React.ComponentClass;

    data: Array<IExampleData<T>>;
}

export interface IComponentExamples<T> {
    name: string;
    component: React.ComponentClass<T>;
    data: T[];
}
