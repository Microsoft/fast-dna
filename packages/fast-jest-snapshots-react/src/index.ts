import * as React from "react";
import * as renderer from "react-test-renderer";

/**
 * An interface describing component example objects used for snapshots and component testing.
 */
export interface ISnapshotTestSuite<T> {
    /**
     * The name of the component
     */
    name: string;

    /**
     * The component constructor
     */
    component: React.ComponentClass<T>;

    /**
     * Prop instances to pass to a component
     */
    data: T[];
}

/**
 * Executes a single snapshot test given a component, component props, and a test title
 */
export function renderSnapshot<T>(props: T, component: React.ComponentClass<T>, title: string): void {
    test(title, (): void => {
        const renderedComponent = renderer.create(React.createElement(component, props));
        const componentJson = renderedComponent.toJSON();

        expect(componentJson).toMatchSnapshot();
    });
}

/**
 * Generate a set of snapshot tests given a snapshot suite
 */
export function generateSnapshots<T>(examples: ISnapshotTestSuite<T>): void {
    const data: T[] = examples.data;
    const component: React.ComponentClass<T> = examples.component;

    if (Array.isArray(data)) {
        data.forEach((example: T, index: number): void => {
            renderSnapshot(example, component, `${examples.name}: ${index}`);
        });
    }
}
