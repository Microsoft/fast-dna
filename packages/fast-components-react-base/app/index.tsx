import * as React from "react";
import * as ReactDOM from "react-dom";
import Site, {
    ISiteProps,
    SiteCategory,
    SiteCategoryIcon,
    SiteCategoryItem
} from "@microsoft/fast-development-site-react";
import * as examples from "./examples";

/**
 * Create the root node
 */
const root: HTMLElement = document.createElement("div");
root.setAttribute("id", "root");
document.body.appendChild(root);

function componentFactory(): JSX.Element[] {
    return Object.keys(examples).map((exampleKey: string) => {
        return (
            <SiteCategory slot={"category"} name={exampleKey}>
                {componentExampleFactory(exampleKey)}
            </SiteCategory>
        );
    });
}

function componentExampleFactory(componentName: string): JSX.Element[] {
    const Component: any = examples[componentName].component;

    return examples[componentName].data.map((componentExample: any) => {
        return (
            <SiteCategoryItem slot={"canvas"}>
                <Component {...componentExample} />
            </SiteCategoryItem>
        );
    });
}

/* tslint:disable */
function render(): void {
    ReactDOM.render(
        <Site title={"FAST components base"}>
            <SiteCategory slot={"category"} name={"Building blocks"}>
                <SiteCategoryIcon slot="category-icon">
                    <svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32">
                        <title>building-blocks</title>
                        <path d="M25.2,16h3V30H2.2V4h14V7l7-7,9,9Zm-21,0h10V6H4.2Zm10,2H4.2V28h10Zm2-7v5h5Zm0,7V28h10V18Zm.47-9,6.53,6.54L29.73,9,23.2,2.46Z"/>
                    </svg>
                </SiteCategoryIcon>
            </SiteCategory>
            <SiteCategory slot={"category"} name={"Components"}>
                {componentFactory()}
            </SiteCategory>
        </Site>,
        root
    );
}

render();
