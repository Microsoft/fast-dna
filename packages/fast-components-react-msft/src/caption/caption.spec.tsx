import * as React from "react";
import * as Adapter from "enzyme-adapter-react-16";
import { configure, shallow } from "enzyme";
import examples from "./examples.data";
import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";
import {
    CaptionLevel,
    CaptionProps,
    CaptionTag,
    ICaptionHandledProps,
    ICaptionManagedClasses,
    ICaptionUnHandledProps
} from "./caption";
import { Typography } from "@microsoft/fast-components-react-base";

/**
 * Configure Enzyme
 */
configure({adapter: new Adapter()});

describe("caption snapshots", (): void => {
    generateSnapshots(examples);
});

describe("caption", (): void => {
    const Component: React.ComponentClass<ICaptionHandledProps> = examples.component;

    test("should render the correct `tag` when `tag` prop is passed", () => {
        const rendered: any = shallow(
            <Component tag={CaptionTag.p} />
        );
        const caption: any = rendered.first().shallow();

        expect(caption.instance().props.tag).toEqual(CaptionTag.p);
    });
});
