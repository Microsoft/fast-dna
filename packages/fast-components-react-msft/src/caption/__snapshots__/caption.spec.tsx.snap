// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`caption snapshots Caption: 0 1`] = `
<p
  className="typography_8-0-1-11 caption-0-1-1 caption_1-0-1-2"
>
  Caption test string
</p>
`;

exports[`caption snapshots Caption: 1 1`] = `
<p
  className="typography_8-0-1-23 caption-0-1-13 caption_1-0-1-14"
>
  Caption test string
</p>
`;
